src/libs/Navigation/linkingConfig/getMatchingCentralPaneRouteForState.ts

```diff
@@ -31,8 +31,45 @@ const getTopMostReportIDFromRHP = (state: State): string => {
     return '';
 };
 
+// Function that checks if the given route has a policyID equal to the id provided in the function params
+function hasRouteMatchingPolicyID(route: NavigationPartialRoute<CentralPaneName>, policyID?: string) {
+    return (
+        route?.params &&
+        'params' in route?.params &&
+        route.params.params &&
+        'policyID' in (route.params.params as Record<string, string | undefined>) &&
+        (route.params.params as Record<string, string | undefined>).policyID === policyID
+    );
+}
+
+// Function that restores already opened screen in the settings tab. Thanks to this function, we are able to open the already selected screen in the workspace settings when we return from another tab.
+function getAlreadyOpenedSettingsScreen(rootState?: State, policyID?: string): keyof CentralPaneNavigatorParamList | undefined {
+    if (!rootState) {
+        return undefined;
+    }
+
+    // If one of the screen from WORKSPACES_SCREENS is now in the navigation state, we can decide which screen we should display.
+    // A screen from the navigation state can be pushed to the navigation state again only if it has a matching policyID with the currently selected workspace.
+    // Otherwise, when we switch the workspace, we want to display the initial screen in the settings tab.
+    const WORKSPACES_SCREENS = TAB_TO_CENTRAL_PANE_MAPPING[SCREENS.WORKSPACE.INITIAL].concat(TAB_TO_CENTRAL_PANE_MAPPING[SCREENS.ALL_SETTINGS]);
+
+    const alreadyOpenedSettingsTab = rootState.routes
+        .filter((item) => item.params && 'screen' in item.params && WORKSPACES_SCREENS.includes(item.params.screen as keyof CentralPaneNavigatorParamList))
```

Comment:
There's a lot of type assertion going on (here and in other places). I gave it a quick look and I couldn't come up with any quick fixes. Do you have any ideas how we could at least reduce the number of type assertions (`something as SomeType`)?